integrationtest/helper.go

```diff
@@ -0,0 +1,163 @@
+package integrationtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/taskcluster/slugid-go/slugid"
+	tcclient "github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/queue"
+	"github.com/taskcluster/taskcluster-worker/commands"
+	_ "github.com/taskcluster/taskcluster-worker/commands/work"
+	_ "github.com/taskcluster/taskcluster-worker/config/env"
+	_ "github.com/taskcluster/taskcluster-worker/config/secrets"
+	_ "github.com/taskcluster/taskcluster-worker/engines/native"
+)
+
+var (
+	// all tests can share taskGroupId so we can view all test tasks in same
+	// graph later for troubleshooting
+	taskGroupID string = slugid.Nice()
+	testdata    string = testdataDir()
+)
+
+func testdataDir() string {
+	// some basic setup...
+	cwd, err := os.Getwd()
+	if err != nil {
+		panic(fmt.Errorf("Could not get current directory during test package initialisation: %v", err))
+	}
+	return filepath.Join(cwd, "testdata")
+}
+
+type TaskPayload struct {
+
+	// Artifacts to be published
+	Artifacts []struct {
+		Expires tcclient.Time `json:"expires,omitempty"`
+
+		// This will be the leading path to directories and the full name
+		// for files that are uploaded to s3. It must not begin or end
+		// with "/" and must only contain printable ascii characters
+		// otherwise.
+		//
+		// Syntax:     ^([\x20-\x2e\x30-\x7e][\x20-\x7e]*)[\x20-\x2e\x30-\x7e]$
+		Name string `json:"name"`
+
+		// File system path of artifact
+		//
+		// Syntax:     ^.*[^/]$
+		Path string `json:"path"`
+
+		// Artifacts can be either an individual `file` or a `directory` containing potentially multiple files with recursively included subdirectories
```

Comment:
Note: the `TaskPayload` definition is generated from the exported json schema, but I can tweak it here.